src/main/java/com/theelixrlabs/healthcare/service/DoctorPatientAssignmentService.java

```diff
@@ -0,0 +1,86 @@
+package com.theelixrlabs.healthcare.service;
+
+import com.theelixrlabs.healthcare.constants.DoctorPatientAssignmentConstants;
+import com.theelixrlabs.healthcare.dao.DoctorPatientAssignmentRepository;
+import com.theelixrlabs.healthcare.dto.DoctorPatientAssignmentDto;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
+import com.theelixrlabs.healthcare.model.DoctorPatientAssignmentModel;
+import com.theelixrlabs.healthcare.repository.DoctorRepository;
+import com.theelixrlabs.healthcare.repository.PatientRepository;
+import com.theelixrlabs.healthcare.utility.MessageUtil;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class DoctorPatientAssignmentService {
+
+    private final DoctorPatientAssignmentRepository doctorPatientAssignmentRepository;
+    private final PatientRepository patientRepository;
+    private final DoctorRepository doctorRepository;
+    private final MessageUtil messageUtil;
+
+    public DoctorPatientAssignmentService(DoctorPatientAssignmentRepository doctorPatientAssignmentRepository,
+                                          PatientRepository patientRepository, DoctorRepository doctorRepository, MessageUtil messageUtil) {
+        this.doctorPatientAssignmentRepository = doctorPatientAssignmentRepository;
+        this.patientRepository = patientRepository;
+        this.doctorRepository = doctorRepository;
+        this.messageUtil = messageUtil;
+    }
+
+    private Optional<DoctorPatientAssignmentModel> validateDoctorPatientAssignment(DoctorPatientAssignmentDto doctorPatientAssignmentDTO) throws CustomException {
```

Comment:
The return type should be void. This method will just validate data and throw the errors.

Reply:
Resolved